lib/tools/system-calls.js

```diff
@@ -0,0 +1,387 @@
+      .replace(/Smart\sDevelopment\sBridge\sversion\s([\d\.]*)[\s\w\-]*/, "$1");
+    let parts = sdbVersion.split('.');
+    return {
+      versionString: sdbVersion,
+      versionFloat: parseFloat(sdbVersion),
+      major: parseInt(parts[0], 10),
+      minor: parseInt(parts[1], 10),
+      patch: parts[2] ? parseInt(parts[2], 10) : undefined,
+    };
+  } catch (e) {
+    log.errorAndThrow(`Error getting sdb version. Original error: '${e.message}'; ` +
+      `Stderr: '${(e.stderr || '').trim()}'; Code: '${e.code}'`);
+  }
+});
+
+systemCallMethods.reboot = async function () {
+  try {
+    let result = await this.shell(["reboot"]);
+    if (result.indexOf("command not found") > -1) {
+      log.debug('Device requires sdb to be running as root in order to reboot. Restarting daemon');
+      await this.root();
+      result = await this.shell(['reboot']);
+      return true;
+    }
+    if (result.indexOf("Rebooting") > -1) {
+      return true;
+    } else {
+      return false;
+    }
+  } catch (err) {
+    log.warn(`Unable to reboot daemon: '${err.message}'.`);
+    return false;
+  }
+};
+
+systemCallMethods.root = async function () {
+  try {
+    await this.sdbExec(['root', 'on']);
+    return true;
+  } catch (err) {
+    log.warn(`Unable to root sdb daemon: '${err.message}'. Continuing`);
+    return false;
+  }
+};
+
+systemCallMethods.unroot = async function () {
+  try {
+    await this.sdbExec(['root', 'off']);
+    return true;
+  } catch (err) {
+    log.warn(`Unable to unroot sdb daemon: '${err.message}'. Continuing`);
+    return false;
+  }
+};
+
+systemCallMethods.fileExists = async function (remotePath) {
+  let files = await this.ls(remotePath);
+  return files.length > 0;
+};
+
+systemCallMethods.ls = async function (remotePath) {
+  try {
+    let stdout = await this.shell(['ls', remotePath]);
+    let lines = stdout.split("\n");
+    return lines.map((l) => l.trim())
+      .filter(Boolean)
+      .filter((l) => l.indexOf("No such file") === -1);
+  } catch (err) {
+    if (err.message.indexOf('No such file or directory') === -1) {
+      throw err;
+    }
+    return [];
+  }
+};
+
+systemCallMethods.getSdkBinaryPath = async function (binaryName) {
+  let binaryLoc = null;
+  let cmd = this.getCommandForOS();
+  try {
+    let { stdout } = await exec(cmd, [binaryName]);
+    log.info(`Using ${binaryName} from ${stdout}`);
+
+    binaryLoc = stdout.trim();
+    return binaryLoc;
+  } catch (e) {
+    log.errorAndThrow(`Could not find ${binaryName} Please set the TIZEN_HOME ` +
+      `environment variable with the Tizen SDK root directory path.`);
+  }
+};
+
+systemCallMethods.getCommandForOS = function () {
+  let cmd = "which";
+  if (system.isWindows()) {
+    cmd = "where";
+  }
+  return cmd;
+};
+
+systemCallMethods.setEmulatorPort = function (emPort) {
+  this.emulatorPort = emPort;
+};
+
+systemCallMethods.installRpm = async function (rpm, path, opts = {}) {
```

Comment:
do we need to provide install/uninstall rpm ?

Reply:
Done